xcm/xcm-builder/src/tests/pay/pay.rs

```diff
@@ -0,0 +1,158 @@
+		AssetKind { destination: (Parent, Parachain(2)).into(), asset_id: Here.into() };
+	let amount = 10 * UNITS;
+
+	new_test_ext().execute_with(|| {
+		// Check starting balance
+		assert_eq!(mock::Assets::balance(0, &recipient), 0);
+
+		assert_ok!(PayOverXcm::<
+			InteriorAccount,
+			TestMessageSender,
+			TestQueryHandler<TestConfig, BlockNumber>,
+			Timeout,
+			AccountId,
+			AssetKind,
+			LocatableAssetKindConverter,
+			AliasesIntoAccountId32<AnyNetwork, AccountId>,
+		>::pay(&recipient, asset_kind, amount));
+
+		let expected_message = Xcm(vec![
+			DescendOrigin(AccountId32 { id: [3u8; 32], network: None }.into()),
+			UnpaidExecution { weight_limit: Unlimited, check_origin: None },
+			SetAppendix(Xcm(vec![ReportError(QueryResponseInfo {
+				destination: (Parent, Parachain(42)).into(),
+				query_id: 1,
+				max_weight: Weight::zero(),
+			})])),
+			TransferAsset {
+				assets: (Here, amount).into(),
+				beneficiary: AccountId32 { id: [5u8; 32], network: None }.into(),
+			},
+		]);
+		let expected_hash = fake_message_hash(&expected_message);
+
+		assert_eq!(
+			sent_xcm(),
+			vec![((Parent, Parachain(2)).into(), expected_message, expected_hash)]
+		);
+
+		let (_, message, hash) = sent_xcm()[0].clone();
+		let message =
+			Xcm::<<XcmConfig as xcm_executor::Config>::RuntimeCall>::from(message.clone());
+
+		// Execute message in parachain 2 with parachain 42's origin
+		let origin = (Parent, Parachain(42));
+		XcmExecutor::<XcmConfig>::execute_xcm(origin, message, hash, Weight::MAX);
+		assert_eq!(mock::Assets::balance(0, &recipient), amount);
+	});
+}
+
+/// Scenario:
+/// A pluralistic body, a Treasury, on the local chain, parachain 42, controls an amount of funds
+/// on parachain 2. [`PayOverXcm::pay`] creates the correct message for the treasury to pay
+/// another account, account #7, on parachain 2, remotely, in the relay's token.
+#[test]
+fn pay_over_xcm_governance_body() {
+	let recipient = AccountId::new([7u8; 32]);
+	let asset_kind =
+		AssetKind { destination: (Parent, Parachain(2)).into(), asset_id: Parent.into() };
+	let amount = 10 * UNITS;
+
+	let relay_asset_index = 1;
+
+	new_test_ext().execute_with(|| {
+		// Check starting balance
+		assert_eq!(mock::Assets::balance(relay_asset_index, &recipient), 0);
+
+		assert_ok!(PayOverXcm::<
+			InteriorBody,
+			TestMessageSender,
+			TestQueryHandler<TestConfig, BlockNumber>,
+			Timeout,
+			AccountId,
+			AssetKind,
+			LocatableAssetKindConverter,
+			AliasesIntoAccountId32<AnyNetwork, AccountId>,
+		>::pay(&recipient, asset_kind, amount));
+
+		let expected_message = Xcm(vec![
+			DescendOrigin(Plurality { id: BodyId::Treasury, part: BodyPart::Voice }.into()),
+			UnpaidExecution { weight_limit: Unlimited, check_origin: None },
+			SetAppendix(Xcm(vec![ReportError(QueryResponseInfo {
+				destination: (Parent, Parachain(42)).into(),
+				query_id: 1,
+				max_weight: Weight::zero(),
+			})])),
+			TransferAsset {
+				assets: (Parent, 10 * UNITS).into(),
```

Comment:
```suggestion
				assets: (Parent, amount).into(),
```

Otherwise, you're gonna have a huge headache trying to figure out why changing `amount` led to an expectation error.

Reply:
I was hardcoding stuff in all the expected values. I'll change them all to variables.